### PR TITLE
AP_Notify: New parameter to enable/disable buzzer

### PR DESCRIPTION
**Pull request by pepevalbe**
This is for people (like me) who are tired of plugging in and out the buzzer while working on the bench.
It have been requested long time ago: https://github.com/ArduPilot/ardupilot/issues/819

I added one more parameter to the AP_Notify group parameter (which I created in other PR).

I hope you find it useful

### CODE REVIEW

File: libraries/AP_Notify/AP_Notify.cpp [hunk 1]
```diff
@@ -41,6 +41,13 @@ const AP_Param::GroupInfo AP_Notify::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("LED_BRIGHT", 0, AP_Notify, _rgb_led_brightness, RGB_LED_HIGH),
 
+    // @Param: BUZZER_ENABLE
+    // @DisplayName: Buzzer enable
+    // @Description: Enable or disable the buzzer in PX4 based boards.
+    // @Values: 0:Disable,1:Enable
+    // @User: Advanced
+    AP_GROUPINFO("BUZZER_EN", 1, AP_Notify, _buzzer_enable, BUZZER_ON),
+
```
Comment (lucasdemarchi) [inline libraries/AP_Notify/AP_Notify.cpp]: nice

File: libraries/AP_Notify/AP_Notify.h [hunk 2]
```diff
@@ -31,10 +31,13 @@
 #define RGB_LED_LOW     1
 #define RGB_LED_MEDIUM  2
 #define RGB_LED_HIGH    3
+#define BUZZER_ON       1
+#define BUZZER_OFF      0
 
 class AP_Notify
 {
     friend class RGBLed;    // RGBLed needs access to notify parameters
+    friend class ToneAlarm_PX4;    // ToneAlarm_PX4 needs access to notify parameters
```
Comment (lucasdemarchi) [inline libraries/AP_Notify/AP_Notify.h]: I think this should be a little bit more generic so it also works on linux.... What about adding a `bool buzzer_enabled() const { return _buzzer_enable:; }` method and avoid the friend class?

File: libraries/AP_Notify/ToneAlarm_PX4.cpp [hunk 3]
```diff
@@ -134,6 +134,11 @@ void ToneAlarm_PX4::update()
         return;
     }
 
+    // exit if buzzer is off
+    if (pNotify->_buzzer_enable == BUZZER_OFF) {
+        return;
+    }
+
```
Comment (lucasdemarchi) [inline libraries/AP_Notify/ToneAlarm_PX4.cpp]: could you do the same in ToneAlarm_Linux.cpp ?

**Comment by rmackay9** [comment]
very nice.
maybe we could name the parameter NOT_BUZZ_ENABLE? or NOT_BUZZER_ENAB?

**Comment by lucasdemarchi** [comment]
@rmackay9 oh.. I already see myself answering to other people if the param enables the buzzer or if it does NOT :)

File: libraries/AP_Notify/ToneAlarm_PX4.cpp (hunk 3 above)
Comment (pepevalbe) [inline libraries/AP_Notify/ToneAlarm_PX4.cpp]: Sure! I will try to do it later.

**Comment by pepevalbe** [comment]
@rmackay9 Since the parameter is in the NTF group its name would be NTF_BUZZER_EN.
I can change it to NTF_BUZZ_ENABLE ot NTF_BUZZER_ENAB. That would be the maximum length, right?

@lucasdemarchi I only did it for ToneAlarm_PX4 because I don't have any Linux based board. I can change it in ToneAlarm_Linux it but I can not verify the result.

File: libraries/AP_Notify/AP_Notify.h (hunk 2 above)
Comment (pepevalbe) [inline libraries/AP_Notify/AP_Notify.h]: Thank you for the advice. I will include this function.

**Comment by pepevalbe** [comment]
I did for ToneAlarm_Linux.cpp as well. Can anyone try it?

I realized that the booting sequence can not be muted this way. Is it a problem?

**Comment by rmackay9** [comment]
I think it's ok if the start-up sequence can't be muted.  It'd be nice if it was but it's a lot tougher to do that.

**Comment by pepevalbe** [comment]
As you can see ToneAlarm_Linux.cpp build failed. 
I have never built for Linux boards. Do you have any clue why it failed?

**Comment by OXINARF** [comment]
The fail was a CI setup problem, nothing to do with your code. It passed all Travis tests, so it should be good to go.

**Comment by pepevalbe** [comment]
Thank you!

**Comment by lucasdemarchi** [comment]
@rmackay9 I'm ok with this. Anything against merging? Otherwise I'll do myself.

**Comment by rmackay9** [comment]
Looks good to me.  I've pushed to master ..

**Pull request closed without merging**